.github/workflows/ci.yml

```diff
@@ -9,90 +9,199 @@ on:
           echo "commit_message=${COMMIT_MSG}" >> $GITHUB_ENV
 
+      # Debug: print the latest commit message
       - name: Debug commit message
         run: |
           echo "The latest commit message is : $commit_message"
 
+      # Validate the commit message
       - name: Validate commit message
         run: |
           ./ci/check_commits.sh "$commit_message"
 
-  run-notebooks:
+  # Testing job (including model training and starting server/client)
+  tests:
     runs-on: ubuntu-latest
     needs: commit-message
 
     steps:
-      # Install Python and Jupyter
+      # Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Set up Python environment
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      # Install project dependencies
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      # Setup dataset (e.g., download, preprocess)
+      - name: Setup dataset
+        run: |
+          set -e
+          make setup-dataset
+
+      # Train the model
+      - name: Train model
+        run: |
+          set -e
+          make train
+
+      # Create necessary directories
+      - name: Create necessary directories
+        run: |
+          mkdir -p models/fhe_files
+          mkdir -p src/fhe_file
+
+      # Run the server in the background
+      - name: Run server in background
+        run: |
+          set -e
+          nohup make run_server > server.log 2>&1 &
+          echo "Waiting for server to start..."
+          sleep 15
+          for i in {1..30}; do
+            if curl -s http://0.0.0.0:8000/docs > /dev/null; then
+              echo "Server is up!"
+              break
+            fi
+            if [ $i -eq 30 ]; then
+              echo "Server failed to start"
+              cat server.log
+              exit 1
+            fi
+            echo "Waiting for server... attempt $i"
+            sleep 1
+          done
+
+      # Run the client in the background
+      - name: Run client in background
+        run: |
+          set -e
+          nohup make run_client > client.log 2>&1 &
+          echo "Waiting for client to start..."
+          sleep 15
+          for i in {1..30}; do
+            if curl -s http://127.0.0.1:8001/docs > /dev/null; then
+              echo "Client is up!"
+              break
+            fi
+            if [ $i -eq 30 ]; then
+              echo "Client failed to start"
+              cat client.log
+              exit 1
+            fi
+            echo "Waiting for client... attempt $i"
+            sleep 1
+          done
+
+      # Run the tests
+      - name: Run tests
+        run: |
+          set -e
+          make test || (echo "Tests failed! Server log:" && cat server.log && echo "Client log:" && cat client.log && exit 1)
```

Comment:
Is it possible to display the accuracy on success?

Reply:
done